src/mito2/src/manifest/tests/checkpoint.rs

```diff
@@ -416,3 +444,57 @@ async fn manifest_install_manifest_to_with_checkpoint() {
     }
     assert!(another_manager.store().total_manifest_size() > 4000);
 }
+
+#[tokio::test]
+async fn test_checkpoint_bypass_in_staging_mode() {
+    common_telemetry::init_default_ut_logging();
+    let (_env, mut manager) = build_manager(1, CompressionType::Uncompressed).await;
+
+    // Apply actions in staging mode - checkpoint should be bypassed
+    for _ in 0..15 {
+        manager
+            .update(
+                nop_action(),
+                RegionRoleState::Leader(RegionLeaderState::Staging),
+            )
+            .await
+            .unwrap();
+    }
+
+    // Wait a bit to ensure no checkpoint is created
+    tokio::time::sleep(std::time::Duration::from_millis(100)).await;
```

Comment:
It does checkpoint in the current thread, no need to sleep.
You can also assert `is_doing_checkpoint()`.